homeassistant/components/sonos/__init__.py

```diff
@@ -396,3 +396,16 @@ async def setup_platforms_and_discovery(self):
                 AVAILABILITY_CHECK_INTERVAL,
             )
         )
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove Sonos config entry from a device."""
+    for identifier in device_entry.identifiers:
+        if identifier[0] != DOMAIN:
+            continue
+        uid = identifier[1]
+        if uid not in hass.data[DATA_SONOS].discovered:
```

Comment:
Do we need to handle manually configured ones or will they be in there?

It would be slightly more efficient to make `hass.data[DATA_SONOS].discovered` a local var so it doesn't have to access the `dict` every loop

Reply:
Manually configured devices will show up in the same way and can be removed as long as they're not actively seen.

Used a local var in [b1d5db3](https://github.com/home-assistant/core/pull/73567/commits/b1d5db3d12f53aa8d027645a945a5bcfb5f2d9dd).